### ISSUE TITLE
Bug: `one-var` complains about const declarations across TS declare module blocks

### ISSUE THREAD

**Issue by TrevorBurnham** [eslint/eslint#15540]
### Environment

Node version: v16.13.0
npm version: v8.1.3
Local ESLint version: v6.8.0 (Currently used)
Global ESLint version: Not found

### What parser are you using?

@typescript-eslint/parser

### What did you do?

[Playground](https://typescript-eslint.io/play/#ts=4.5.2&sourceType=module&code=CYUwxgNghgTiAEBbA9sArhBByAVAOgCsAHAcy3gG8AoeeMZAOwGcAXOxlkBlgLnlZgBLBiQDcNeCAAeRZDDagAZlAxt63Li3EBfKlVCRYCFOkzxchIiDKUJ61uw3c+A4WInTZ8+EpUQ1HJo6QA&rules=N4Ig9gdgpgtAbgQwE4gFwikpYUF8g&tsConfig=N4XyA)

<details>
<summary>Configuration</summary>

{
  "plugins": ["@typescript-eslint"],
  "parser": ["@typescript-eslint/parser"],
  "rules": {
    "one-var": ["error"]
  }
}
</details>

```ts
declare module '*.jpg' {
  const content: string;
  export default content;
}

declare module '*.jpeg' {
  const content: string;
  export default content;
}
```

### What did you expect to happen?

The `one-var` rule should not be triggered. The two `const` declarations are in different `declare module` blocks, and therefore can't be combined.

### What actually happened?

The second `const` declaration is reported as a `one-var` rule violation:

```
Combine this with the previous 'const' statement.
```

### Participation

- [X] I am willing to submit a pull request for this issue.

### Additional comments

I've filed a downstream issue on `typescript-eslint` as well, but it sounds like this needs to be addressed in the `one-var` rule itself: https://github.com/typescript-eslint/typescript-eslint/issues/4475

**Comment by bradzacher** [comment]
It appears that this rule makes two assumptions:
1) that the only variable scopes that can only ever exist are those that it knows about.
2) that a `const`/`let` variable that has two declarations is an error.

I feel like this rule should be updated to use scope analysis scopes so that (1) can correctly function in language constructs that ESLint doesn't yet support (eg I noticed that the rule had to be updated for `StaticBlock` - which could have been avoided if scope analysis scopes had been used to identify the new variable scope automatically).

(2) is probably fine to keep in the rule (though most parsers / tooling will error on it anyway), but it's probably worth evaluating to determine if it's worth flagging in the rule at all. Note that duplicate let/const is a parser error in espree.

**Comment by nzakas** [comment]
Thanks for the input. As [stated in the README](https://github.com/eslint/eslint#stylistic-rule-updates), we actually aren't making any further updates to rules that enforce stylistic preferences. If you have a strong preference for such a change, you can always copy the existing rule into a custom rule and make it behave the way you want.

(Stylistic rules necessarily can’t have any guarantees about non-JavaScript languages.)
